GMG/Core/Utils/View+Extensions.swift

```diff
@@ -13,12 +13,19 @@ extension View {
 extension View {
     func navigationBar<Leading: View, Trailing: View>(
         _ title: String? = nil,
+        isTitleEditable: Bool = false,
+        onEnterTitle: ((String) -> Void)? = nil,
         isBackButtonHidden: Bool = false,
         @ViewBuilder leading: () -> Leading,
         @ViewBuilder trailing: () -> Trailing
     ) -> some View {
         VStack {
-            NavigationBar(title) {
+            NavigationBar(
+                title,
+                isBackButtonHidden: isBackButtonHidden,
+                isTitleEditable: isTitleEditable,
+                onEnterTitle: onEnterTitle
+            ) {
                 leading()
             } trailing: {
                 trailing()
```

Comment:
`NavigationBar` 자체에 타이틀 변경 로직을 넣기보다는 `Center`로 뷰를 넘겨주는 식으로 하는 것이 어떨까요? 확장성 면에서 더 좋을 것 같습니다!

```suggestion
extension View {
    func navigationBar<Leading: View, Center: View, Trailing: View>(
        _ title: String? = nil,
        isBackButtonHidden: Bool = false,
        @ViewBuilder leading: () -> Leading,
        @ViewBuilder center: () -> Center,
        @ViewBuilder trailing: () -> Trailing
    ) -> some View {
    VStack {
        NavigationBar(
            title,
            isBackButtonHidden: isBackButtonHidden,
            isTitleEditable: isTitleEditable,
            onEnterTitle: onEnterTitle
        ) {
            leading()
        } center: {
            center()
        } trailing: {
            trailing()
```

Reply:
네 확장성 말고도, 통일성 측면에서도 leading이랑 trailing 처럼 컴포넌트를 인자로 받는게 좋겠네요!

Reply:
네 확장성 말고도, 통일성 측면에서도 leading이랑 trailing 처럼 컴포넌트를 인자로 받는게 좋겠네요!